### PR TITLE
modules: openthread: platform: radio: Time Sync OT Feature support

### PR DESCRIPTION
**Pull request by ghost**
Add missing otRadioIeInfo structure and add plaform time to transmit frame

### CODE REVIEW

File: modules/openthread/platform/radio.c [hunk 1]
```diff
@@ -385,6 +393,25 @@ void transmit_message(struct k_work *tx_job)
...
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
+	if (sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0) {
+		*(sTransmitFrame.mPsdu + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset) =
+			sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeSyncSeq;
+		uint8_t *time_ie =
+			sTransmitFrame.mPsdu + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset;
+		uint64_t offset_plat_time =
+			otPlatTimeGet() + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mNetworkTimeOffset;
+
+		*time_ie = sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeSyncSeq;
```
Comment (rlubos) [inline modules/openthread/platform/radio.c:405]: Doesn't this duplicate the write at line 398 (I like this approach better though)?

File: modules/openthread/platform/radio.c [hunk 2]
```diff
@@ -385,6 +393,25 @@ void transmit_message(struct k_work *tx_job)
...
+
+		*(++time_ie) = (uint8_t)(offset_plat_time & 0xff);
+		for (uint8_t i = 1; i < sizeof(uint64_t); i++) {
+			offset_plat_time = offset_plat_time >> 8;
+			*(++time_ie) = (uint8_t)(offset_plat_time & 0xff);
+		}
```
Comment (rlubos) [inline modules/openthread/platform/radio.c:411]: Are we just writing 64-bit integer to the buffer in little endian here? We have `sys_put_le64()` helper in Zephyr (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/byteorder.h#L485).

Comment (DaveLacerteHQ) [reply modules/openthread/platform/radio.c]: Changed for sys_put_le64. Thanks for the hint. Way cleaner!

File: modules/openthread/platform/radio.c [hunk 3]
```diff
@@ -385,6 +393,20 @@ void transmit_message(struct k_work *tx_job)
 	radio_api->set_channel(radio_dev, channel);
 	radio_api->set_txpower(radio_dev, get_transmit_power_for_channel(channel));
 
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
+	if (sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0) {
+		*(sTransmitFrame.mPsdu + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset) =
```
Comment (rlubos) [inline modules/openthread/platform/radio.c:398]: @DaveLacerteHQ In my [previous comment](https://github.com/zephyrproject-rtos/zephyr/pull/70758#discussion_r1540664551) I was referring to this line, not the line below where the pointer is initialized.

Comment (DaveLacerteHQ) [reply modules/openthread/platform/radio.c]: You are right. Sorry for the confusion. Corrected.

**Comment by canisLupus1313** [comment]
@DaveLacerteHQ  Time sync is experimental feature of Openthread and not a part of Thread specification. Would be nice if you could add `EXPERIMANTAL` to `OPENTHREAD_TIME_SYNC`. 
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/openthread/Kconfig.features#L343

**Review by rlubos — DISMISSED** [review]
Techically looks good, but would be good to get second review from someone more familiar with those areas of OT stack.

**Comment by DaveLacerteHQ** [comment]
> @DaveLacerteHQ Time sync is experimental feature of Openthread and not a part of Thread specification. Would be nice if you could add `EXPERIMANTAL` to `OPENTHREAD_TIME_SYNC`. https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/openthread/Kconfig.features#L343

Done

**Review by canisLupus1313 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**